### PR TITLE
add the `name_override` directive to allow names to be programmatical…

### PR DESCRIPTION
**Pull request by nh13**
…ly set

### CODE REVIEW

File: tests/test_name_override/Snakefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+rule all:
+    input:
+        txt = 'out2.txt'
+
+some_var = 'generate_out'
+
+rule:
+    name_override: some_var
```
Comment (johanneskoester) [inline tests/test_name_override/Snakefile:8]: I think I'd just call this `name:`.

File: snakemake/parser.py [hunk 2]
```diff
@@ -396,6 +396,12 @@ def block_content(self, token):
 # Rule keyword states
 
 
+class NameOverride(RuleKeywordState):
```
Comment (johanneskoester) [inline snakemake/parser.py:399]:
```suggestion
class Name(RuleKeywordState):
```

File: snakemake/parser.py [hunk 3]
```diff
@@ -640,6 +646,7 @@ def args(self):
 
 class Rule(GlobalKeywordState):
     subautomata = dict(
+        name_override=NameOverride,
```
Comment (johanneskoester) [inline snakemake/parser.py:649]:
```suggestion
        name=NameOverride,
```

**Review by johanneskoester — CHANGES_REQUESTED** [review]
brilliant idea! Just one thing.

File: tests/test_name_override/Snakefile (hunk 1 above)
Comment (nh13) [reply tests/test_name_override/Snakefile]: Done!

File: snakemake/parser.py (hunk 3 above)
Comment (nh13) [reply snakemake/parser.py]: Done

File: snakemake/parser.py (hunk 2 above)
Comment (nh13) [reply snakemake/parser.py]: Done

**Comment by nh13** [comment]
@johanneskoester made it work with `name` instead of `name_override` :)

**Review by johanneskoester — APPROVED** [review]

**Comment by johanneskoester** [comment]
Thank you so much!

**Pull request merged**